### DISCUSSION TITLE
Understand how ValidDatatypeLinks are defined

### DISCUSSION THREAD

**Discussion by rnewell311** [codatio/codat-docs#297] [Q&A] [answered]
Hi all,

I have a question about the `ValidDatatypesLinks` and how they are defined. We have a customer request to map Asset accounts to a bill line item in their ERP, and as we have been looking into it we have noticed some weird discrepancies in the `ValidDatatypeLinks`. For example, we have two asset accounts in QBO, where one includes `Bill.LineItems.AccountRef.Id` in the ValidDatatypesLinks. However, in the second asset account the `ValidDatatypesLinks` is blank. I imagine this is something configured in the ERP, but could you give me an idea how the ValidDatatypesLinks are defined so we can understand what's going on and if/how we could address the customer need? Thanks!
![image](https://user-images.githubusercontent.com/11344147/230130193-b19ded98-4d79-4c23-85b8-39871f784673.png)
![image (1)](https://user-images.githubusercontent.com/11344147/230130231-9eeb308e-a038-488e-991b-8b9abc2b0583.png)

**Comment by dcoplowe** [comment, marked as answer]
Hi rnewell311, 

Valid data type links are dependent on the `fullyQualifiedCategory` and not the account `type`. For the account named 'checking', its `fullyQualifiedCategory` is `Asset.Bank.Checking` whereas the account named 'Accounts Receivable (A/R)' is a `Assess.Account Receivable.AccountsReceivable`. Now there are a number of ways we define valid links for QBO based on the `fullyQualifiedCategory`. For example

- If the `fullyQualifiedCategory`  contains `Bank` or `Other Current Asset` then a Payment is a valid link.
- If the `fullyQualifiedCategory`  contains `Bank` or `Credit Card` then a Bill Payment is a valid link.
- If the `fullyQualifiedCategory` does **not** contain `Accounts Receivable` or `Accounts Payable` then Bill, Bill Credit Note, Journal Entry & Item is a valid link.

This is why we do not see any available `validDataTypeLinks` for the second account named 'Accounts Receivable (A/R)'. 

More information on valid data type links can be found [here](https://docs.codat.io/accounting-api#/schemas/ValidDataTypeLinks). Note that availability and mappings of valid data type links are different per integration.

Many thanks,

David
